sound/soc/sof/ipc4-topology.c

```diff
@@ -1713,14 +1713,20 @@ sof_ipc4_prepare_copier_module(struct snd_sof_widget *swidget,
 			 */
 			i = 0;
 			list_for_each_entry(w, &sdev->widget_list, list) {
+				u32 node_type;
+
 				if (w->widget->sname &&
 				    strcmp(w->widget->sname, swidget->widget->sname))
 					continue;
 
 				dai = w->private;
 				alh_copier = (struct sof_ipc4_copier *)dai->private;
 				alh_data = &alh_copier->data;
-				blob->alh_cfg.mapping[i].device = alh_data->gtw_cfg.node_id;
+				node_type = SOF_IPC4_GET_NODE_TYPE(alh_data->gtw_cfg.node_id);
+				blob->alh_cfg.mapping[i].device =
+					SOF_IPC4_GET_NODE_TYPE(node_type);
```

Comment:
Shouldn't this be `SOF_IPC4_NODE_TYPE(node_type);`?

Reply:
well spotted @bardliao that would indeed shift the initial value twice.

@ranj063 what do you think?

Reply:
@bardliao thats correct. Let me fix this

Reply:
@bardliao @plbossart fixed in #4829